### ISSUE TITLE
Less: CssSyntaxError "Unknown word" for "each"

### ISSUE THREAD

**Issue by Xotic750** [prettier/prettier#5653]
**Environments:**
- Prettier Version: 1.14.3
- Running Prettier via: CLI, Node.js
- Runtime:  Node.js v8
- Operating System:  macOS

**Steps to reproduce:**

From issue raised with Stylelint https://github.com/stylelint/stylelint/issues/3854

> > Clearly describe the bug
> 
> It's not clear to me in which package this problem is being caused, but when I add _stylelint-prettier_ and enable the `'prettier/prettier': true,` then stylelint reports `each` as `CssSyntaxError Unknown word`.
> 
> > Which rule, if any, is the bug related to?
> 
> `prettier/prettier`
> 
> > What CSS is needed to reproduce the bug?
> 
> ```less
> @selectors-color: primary, secondary, info, success, warning, error, dark, light;
> 
> @primary: #007bff;
> @secondary: #6c757d;
> @info: #17a2b8;
> @success: #28a745;
> @warning: #ffc107;
> @error: #dc3544;
> @dark: #343a40;
> @light: #f8f9fa;
> 
> each(@selectors-color, {
>   .bg-@{value} {
>     background-color: @@value;
>   }
> });
> ```
> 
> > What stylelint configuration is needed to reproduce the bug?
> 
> ```js
> /**
>  * @file Manages the root configuration settings for project wide stylelint.
>  * @module stylelint/root/configuration
>  * @version 1.0.0
>  * @see {@link https://stylelint.io/} for further information.
>  */
> 
> module.exports = {
>   /**
>    * Extend an existing configuration.
>    * @type {string|Array.<string>}
>    * @see {@link https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#extends}
>    */
>   extends: [
>     /**
>      * The standard shareable config for stylelint.
>      * @type {string}
>      * @see {@link https://github.com/stylelint/stylelint-config-standard}
>      */
>     'stylelint-config-standard',
>     /**
>      * Turns off all rules that are unnecessary or might conflict with prettier.
>      * @type {string}
>      * @see {@link https://github.com/shannonmoeller/stylelint-config-prettier}
>      */
>     'stylelint-config-prettier',
>   ],
> 
>   /**
>    * @type {Array.<string>}
>    */
>   plugins: [
>     'stylelint-prettier',
>     'stylelint-no-unsupported-browser-features',
>     'stylelint-declaration-strict-value',
>   ],
> 
>   /**
>    * @type {Array.<string>}
>    */
>   processors: [], // do not include empty
> 
>   /**
>    * @type {!Object}
>    */
>   rules: {
>     'no-empty-source': null,
>     'no-missing-end-of-source-newline': null, // currently has a bug #3428
>     'prettier/prettier': true,
>     'plugin/no-unsupported-browser-features': [true, {
>       severity: 'warning',
>     }],
>     'scale-unlimited/declaration-strict-value': ['color', {
>       ignoreKeywords: 'inherit',
>     }],
>   },
> };
> ```
> 
> **postcss.config.js**
> ```js
> const postcssHTML = require('postcss-html');
> const postcssURL = require('postcss-url');
> const postcssSafe = require('postcss-safe-parser');
> const postcssAutoprefixer = require('autoprefixer');
> 
> module.exports = {
>   plugins:  [postcssHTML, postcssURL, postcssSafe, postcssAutoprefixer],
> };
> ```
> > Which version of stylelint are you using?
> 
> ```json
>   "devDependencies": {
>     "less": "^3.9.0",
>     "postcss": "^7.0.7",
>     "postcss-cssnext": "^3.1.0",
>     "postcss-html": "^0.34.0",
>     "postcss-less": "^3.1.0",
>     "postcss-loader": "^3.0.0",
>     "postcss-safe-parser": "^4.0.1",
>     "postcss-sass": "^0.3.5",
>     "postcss-scss": "^2.0.0",
>     "postcss-syntax": "^0.34.0",
>     "postcss-url": "^8.0.0",
>     "prettier": "^1.14.3",
>     "stylelint": "^9.9.0",
>     "stylelint-config-prettier": "^4.0.0",
>     "stylelint-config-standard": "^18.2.0",
>     "stylelint-declaration-strict-value": "^1.1.2",
>     "stylelint-formatter-pretty": "^1.0.3",
>     "stylelint-no-unsupported-browser-features": "^3.0.2",
>     "stylelint-prettier": "^1.0.5",
>   },
> ```
> > How are you running stylelint: CLI, PostCSS plugin, Node API?
> 
> CLI from _package.json_ with `"lint:css": "stylelint '**/test.{css,less,vue}'"`
> 
> > Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?
> 
> Yes, it's related to LESS `each`
> 
> > What did you expect to happen?
> 
> No warnings or error to be flagged.
> 
> > What actually happened (e.g. what warnings or errors did you get)?
> 
> ```shell
> > stylelint '**/test.{css,less,vue}'
> 
> SyntaxError: (postcss) CssSyntaxError Unknown word (12:1)
>   10 | @light: #f8f9fa;
>   11 | 
> > 12 | each(@selectors-color, {
>      | ^
>   13 |   .bg-@{value} {
>   14 |     background-color: @@value;
>   15 |   }
>     at createError (/Users/grahamfairweather/source/i-view-core-x/node_modules/prettier/parser-postcss.js:19093:15)
>     at parseWithParser (/Users/grahamfairweather/source/i-view-core-x/node_modules/prettier/parser-postcss.js:19001:11)
>     at Object.parse (/Users/grahamfairweather/source/i-view-core-x/node_modules/prettier/parser-postcss.js:19036:12)
>     at Object.parse$2 [as parse] (/Users/grahamfairweather/source/i-view-core-x/node_modules/prettier/index.js:7138:19)
>     at coreFormat (/Users/grahamfairweather/source/i-view-core-x/node_modules/prettier/index.js:10398:23)
>     at format (/Users/grahamfairweather/source/i-view-core-x/node_modules/prettier/index.js:10570:16)
>     at formatWithCursor (/Users/grahamfairweather/source/i-view-core-x/node_modules/prettier/index.js:10582:12)
>     at /Users/grahamfairweather/source/i-view-core-x/node_modules/prettier/index.js:34924:15
>     at Object.format (/Users/grahamfairweather/source/i-view-core-x/node_modules/prettier/index.js:34943:12)
>     at /Users/grahamfairweather/source/i-view-core-x/node_modules/stylelint-prettier/stylelint-prettier.js:77:39
> ```

Issue was closed https://github.com/stylelint/stylelint/issues/3854#issuecomment-447981053

> It's an error in prettier, which uses an old version of the postcss-less parser, which doesn't support each.

**Comment by j-f1** [comment]
Repro:

**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABAznANnMMIBO6AtJFoUgAQAOBAlgLYCGBAngDSWaRQAmLHSnSgAzCJ3QBXMGDjp0nAO4sowgOac4BAoU78CAa05Y6agBYwA3AB0ot1LUYCqAYgAMbgOwAjESJtQGLjQ+qyuAGxgngCsnrwBqMJirgCMnkwATN4AHAlSMnLorhnZTJ4ALNEJygSqUGqufmApXglaOgSuvGAAzNHl5Qn6Bq495T1M5W4JJuYwjdkiAJwiTAG2cExgZgAUQTh4hCRkupTAtpSUAHTeasSowABuTFiScAC+ZxeXlN5bBmodJI+KQIOROpRUKhnq84AFLu9bO8AJQBEDsEAQagwOjQdDIUAsHSKAAKLAQ+JQTEeEDovHRIG8BH+cBgAGVqFt1MgYAQ3hiLAwsAB1Mx0eDoTmyNkU8V0R7i1jIcDyBnCTAEGAk5lqZjIVZYTAYgBW6AAHgAhZlgAystlMBhwAAywjg+peRpAprNbPUOAAipIIPB3Ya4BjOUQtMqcKqI-RYMK6TAzMhsm54xBMMLmdRlbQ5FpHm6MQQ4ABHSR0Mvapi6pihz2YBh0Hl88MgdB+uCB4NupAGz0wJjeJO8FPIDIY3lMOgmeoAYQgDD1KCg0BLIEkmAAKiPKYOPu8gA)
```sh
--parser less
```

**Input:**
```less
@selectors-color: primary, secondary, info, success, warning, error, dark, light;

@primary: #007bff;
@secondary: #6c757d;
@info: #17a2b8;
@success: #28a745;
@warning: #ffc107;
@error: #dc3544;
@dark: #343a40;
@light: #f8f9fa;

each(@selectors-color, {
  .bg-@{value} {
    background-color: @@value;
  }
});

```

**Output:**
```less
SyntaxError: (postcss) CssSyntaxError Unknown word (12:1)
  10 | @light: #f8f9fa;
  11 | 
> 12 | each(@selectors-color, {
     | ^
  13 |   .bg-@{value} {
  14 |     background-color: @@value;
  15 |   }
```

cc @evilebottnawi

**Comment by Xotic750** [comment]
When this is fixed would it be possible to get this in 1.14 too, as the formatting changes that come with 1.15 are too much w.r.t. our _.vue_ project presently?

**Comment by j-f1** [comment]
I don’t think we provide support for old versions of Prettier unfortunately.

**Comment by Xotic750** [comment]
Ok, it was worth an ask.

**Comment by yoannmoinet** [comment]
Also have this issue, is there any known work-around?

**Comment by Xotic750** [comment]
I have tried updating the dependency `postcss-less` to v3.1.4. A change in code is needed to point to the parser `const LessParser = require("postcss-less/dist/less-parser");` -> `const LessParser = require("postcss-less/lib/lessParser");` one of the major difference that I see when running the tests is the form used for comments goes from `//` to `/* */`, which is no biggy. However I see more serious changes, things like

```
      pre {
    -   .wrap;
    +   @wrap;
      }
```
```      
      .class {
    -   #my-library > .my-mixin();
    +   @my-library > .my-mixin();
      }
```
```
   - @variable: .selector;
    + @variable : .selector .selector;
```
So it's not just a simple update of the package.

**Comment by xdlrt** [comment]
so far no solution for this problem?

**Comment by evilebottnawi** [comment]
Here solution https://github.com/prettier/prettier/pull/6981

**Comment by xdlrt** [comment]
> Here solution #6981

thanks, hope merge this pr sooner :)

**Comment by thorn0** [comment]
fixed by #6981
